packages/@uppy/xhr-upload/src/index.js

```diff
@@ -214,24 +215,20 @@ module.exports = class XHRUpload extends Plugin {
     this.addMetadata(formPost, meta, opts)
 
     files.forEach((file) => {
-      const opts = this.getOptions(file)
+      const options = this.getOptions(file)
 
       const dataWithUpdatedType = setTypeInBlob(file)
 
       if (file.name) {
-        formPost.append(opts.fieldName, dataWithUpdatedType, file.name)
+        formPost.append(options.fieldName, dataWithUpdatedType, file.name)
       } else {
-        formPost.append(opts.fieldName, dataWithUpdatedType)
+        formPost.append(options.fieldName, dataWithUpdatedType)
       }
     })
 
     return formPost
   }
 
-  createBareUpload (file, opts) {
-    return file.data
-  }
```

Comment:
Removing this is technically a breaking change, and that's a strong case to switch to private syntax (`#createBareUpload`).